Setup simple Github Actions

rosik opened the pull request:
Part of #2

Totktonada commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,37 @@
+name: Publish
```
If you don't mind, I would apply all comments from https://github.com/tarantool/smtp/pull/38 here.

Totktonada replied:
The idea about uploading .src.rock is nice. So implemented it for the smtp module in https://github.com/tarantool/smtp/pull/41 and I'll keep it here.

Totktonada replied:
Done.

Totktonada commented on test/CMakeLists.txt:
```diff
@@ -1,2 +1,5 @@
+add_test(
+    NAME tuple_keydef.test.lua
+    COMMAND runtest.sh tuple_keydef.test.lua
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
```
If you don't mind, I'll move it into its own commit. [It seems](https://github.com/tarantool/tuple-keydef/pull/10#discussion_r639017243), the change fixes test running, when out-of-source build is used, and, strictly speaking, is neither about CI, nor about hot reload.

Totktonada replied:
Moved to PR #12.

Totktonada commented on .github/workflows/test-fast.yml:
```diff
@@ -0,0 +1,26 @@
+      matrix:
+        tarantool:
+          - '1.10'
+          - '2.5'
```
I'll add 2.4, because necessary APIs are available starting from 2.4.3.

Totktonada commented on .github/workflows/test-fast.yml:
```diff
@@ -0,0 +1,26 @@
+name: Test
+
+on:
+  push:
+  workflow_dispatch:
```
I'll add `pull_request` here and will [skip duplicate jobs](https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012/6#M9370). So, external PRs will not be a kind on second class citizens.

rosik committed:
dev: gitignore 'tarantoolctl rocks make' artifacts

It eases local experimenting around rocks.

Co-authored-by: Alexander Turenko <alexander.turenko@tarantool.org>

Part of #2

rosik committed:
ci: add simple fast testing on GitHub Actions

Unlike the grand plan with different Linux distros / Mac OS, RPM / Deb
packages building and verification, past and actual tarantool releases,
this workflow just runs testing on several latest tarantool versions on
Ubuntu Linux.

However it uses the great setup-tarantool GitHub Action and so
incredibly fast due to caching of tarantool and its dependencies. This
workflow will give us fast CI feedback.

Another nice property is that it verifies a flow similar to the user's
one (`tarantoolctl rocks install <...>`). A usual development flow like
`cmake . && make && make test` may miss an installation problem.

Co-authored-by: Alexander Turenko <alexander.turenko@tarantool.org>

Part of #2

rosik committed:
ci: publish rockspec and .src.rock

The new worklow uploads the scm-1 rockspec on pushing a commit to master
and uploads the release rockspec + .src.rock tarball at pushing a tag.

This automation descrease amount of manual work to release the module.

Co-authored-by: Alexander Turenko <alexander.turenko@tarantool.org>

Part of #2

Totktonada reviewed (approved):
LGTM.

I splitted the changes to separate commits, added brief description of the changes, rearranged the code a bit (I asked Yaroslav and he didn't complain). That's mostly to make me more comfortable around the module's code. (How far it from madness? I don't know).

I'll wait for #12 and Yaroslav's agreement about my changes prior to merge this PR.

Totktonada commented:
(That's fast as a rocket, I can't stop admire!)

Totktonada commented:
Rebased after PR #12.
